base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorViewModel.java

```diff
@@ -398,7 +398,17 @@ protected void insertEmptyAction(Integer insertionLocation) {
      *             the actions to delete.
      */
     protected void deleteAction(List<ScriptGeneratorAction> actionsToDelete) {
+    int selectedIndex = viewTable.getSelectionIndex();
+    // Exclude the last empty row
+    int actionCount = viewTable.table().getItemCount() - 1;
+    // If last action is selected, select new last action, otherwise select next action
+    int toSelect = selectedIndex == actionCount - 1 ? selectedIndex - actionsToDelete.size() : selectedIndex - actionsToDelete.size() + 1;
+    
```

Comment:
you will also need to take into account that user could select empty row with non empty rows and try to delete the rows.